resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProviderSuite.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes
+
+import java.io.FileInputStream
+import java.nio.file.Files
+import java.security.KeyStore
+import javax.net.ssl.SSLContext
+
+import com.google.common.base.Charsets
+import com.google.common.io.{BaseEncoding, Files => GuavaFiles}
+import io.fabric8.kubernetes.api.model.{DoneableSecret, EnvVar, EnvVarBuilder, Secret, SecretBuilder, SecretList, Volume, VolumeBuilder, VolumeMount, VolumeMountBuilder}
+import io.fabric8.kubernetes.client.KubernetesClient
+import io.fabric8.kubernetes.client.dsl.{MixedOperation, Resource}
+import org.mockito.Matchers.any
+import org.mockito.Mockito._
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.scalatest.BeforeAndAfter
+import org.scalatest.mock.MockitoSugar.{mock => simpleMock}
+
+import org.apache.spark.{SparkConf, SparkFunSuite}
+import org.apache.spark.deploy.kubernetes.config._
+import org.apache.spark.deploy.kubernetes.constants._
+import org.apache.spark.util.Utils
+
+private[spark] class SslConfigurationProviderSuite extends SparkFunSuite with BeforeAndAfter {
+
+  private type SecretResource = Resource[Secret, DoneableSecret]
+  private type SecretsHandler = MixedOperation[Secret, SecretList, DoneableSecret, SecretResource]
+
+  private val APP_ID = "app-id"
+  private val KEYSTORE_PASSWORD = "keystore"
+  private val KEY_PASSWORD = "key"
+  private val TRUSTSTORE_PASSWORD = "truststore"
+  private val IP_ADDRESS = "192.168.99.100"
+  private val SSL_SECRET_DIR = s"$DRIVER_CONTAINER_SECRETS_BASE_DIR/$APP_ID-ssl"
+
+  private val sslFolder = Files.createTempDirectory("ssl-configuration-provider-suite").toFile
```

Comment:
The current approach looks fine to me. I see many tests in `core` uses `Utils.createTempDir`, which is similar to this code:
```
  /**
   * Create a temporary directory inside the given parent directory. The directory will be
   * automatically deleted when the VM shuts down.
   */
  def createTempDir(
      root: String = System.getProperty("java.io.tmpdir"),
      namePrefix: String = "spark"): File = {
    val dir = createDirectory(root, namePrefix)
    ShutdownHookManager.registerShutdownDeleteDir(dir)
    dir
  }
```

Here's code snippet from [SparkContextSuite](https://github.com/apache-spark-on-k8s/spark/blob/k8s-support-alternate-incremental/core/src/test/scala/org/apache/spark/SparkContextSuite.scala):
```
  test("basic case for addFile and listFiles") {
    val dir = Utils.createTempDir()

    val file1 = File.createTempFile("someprefix1", "somesuffix1", dir)
    val absolutePath1 = file1.getAbsolutePath
```

Maybe we want to switch to this to be consistent. But, I don't have a strong preference here. i.e. I think it's your call.